pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -57,17 +61,20 @@ def __init__(self, monitor: str = 'val_loss', min_delta: float = 0.0, patience:
         self.wait = 0
         self.stopped_epoch = 0
         self.mode = mode
-        self.mode_dict = {
-            'min': torch.lt,
-            'max': torch.gt,
-            'auto': torch.gt if 'acc' in self.monitor else torch.lt
-        }
 
         if mode not in self.mode_dict:
             if self.verbose > 0:
                 log.info(f'EarlyStopping mode {mode} is unknown, fallback to auto mode.')
             self.mode = 'auto'
 
+        if self.mode == 'auto':
+            if self.monitor == 'acc':
```

Comment:
I guess that there are more metrics which are auto max, not only acc, for example, precision, f1...
maybe add a todo to be updated with new metrics package

Reply:
fwiw i think we should just ask the user explicitly (as a compromise i at least added logging to alert the user what we're doing)

Reply:
so you mean completely drop `auto`?

Reply:
yeah, although i wouldn't want to drop it if it'd be disruptive to our users